### PR TITLE
Sync: Send home and siteurl along with sync data

### PR DESCRIPTION
**Pull request by dereksmart**
This will open up options for how we can check for IDC server-side.

cc @ebinnion @roccotripaldi

### CODE REVIEW

File: sync/class.jetpack-sync-actions.php [hunk 1]
```diff
@@ -124,6 +124,8 @@ static function send_data( $data, $codec_name, $sent_timestamp, $queue_id ) {
 			'codec'     => $codec_name, // send the name of the codec used to encode the data
 			'timestamp' => $sent_timestamp, // send current server time so we can compensate for clock differences
 			'queue'     => $queue_id, // sync or full_sync
+			'home'      => get_option( 'home' ), // Send home url option to check for Identity Crisis server-side
```
Comment (kraftbj) [inline sync/class.jetpack-sync-actions.php]: Should we send both the option value and the function results `get_home_url`, `get_site_url`?

Comment (dereksmart) [inline sync/class.jetpack-sync-actions.php]: Yeah solid to account for filters and whatnot

File: jetpack.php [hunk 2]
```diff
@@ -23,7 +23,7 @@
 defined( 'JETPACK_CLIENT__AUTH_LOCATION' )   or define( 'JETPACK_CLIENT__AUTH_LOCATION', 'header' );
 defined( 'JETPACK_CLIENT__HTTPS' )           or define( 'JETPACK_CLIENT__HTTPS', 'AUTO' );
 defined( 'JETPACK__GLOTPRESS_LOCALES_PATH' ) or define( 'JETPACK__GLOTPRESS_LOCALES_PATH', JETPACK__PLUGIN_DIR . 'locales.php' );
-defined( 'JETPACK__API_BASE' )               or define( 'JETPACK__API_BASE', 'https://jetpack.wordpress.com/jetpack.' );
+defined( 'JETPACK__API_BASE' )               or define( 'JETPACK__API_BASE', 'https://dsmartsandbox.wordpress.com/jetpack.' );
```
Comment (ebinnion) [inline jetpack.php]: :trollface:

**Comment by dereksmart** [comment]
closing in favor of #5278

**Pull request closed without merging**